bip-schnorr/test-vectors.py

```diff
@@ -23,48 +23,48 @@ def vector1():
     pubkey_point = point_from_bytes(pubkey)
     assert(pubkey_point[1] & 1 == 1)
 
-    return (bytes_from_int(seckey), pubkey, msg, sig, "TRUE", None)
+    return (seckey, pubkey, msg, sig, "TRUE", None)
 
 def vector2():
-    seckey = 0xC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B14E5C9
+    seckey = bytes_from_int(0xC90FDAA22168C234C4C6628B80DC1CD129024E088A67CC74020BBEA63B14E5C9)
     msg = bytes_from_int(0x5E2D58D8B3BCDF1ABADEC7829054F90DDA9805AAB56C77333024B9D0A508B75C)
     sig = schnorr_sign(msg, seckey)
 
-    # This singature vector would not verify if the implementer checked the
-    # jacobi symbol of the X coordinate of R instead of the Y coordinate.
+    # This test vector would not pass verification if the implementer checked
+    # is_square of the X coordinate of R instead of the Y coordinate.
     R = point_from_bytes(sig[0:32])
-    assert(jacobi(R[0]) != 1)
+    assert(not is_square(R[0]))
 
-    return (bytes_from_int(seckey), pubkey_gen(seckey), msg, sig, "TRUE", None)
+    return (seckey, pubkey_gen(seckey), msg, sig, "TRUE", None)
 
 def vector3():
-    seckey = 0x0B432B2677937381AEF05BB02A66ECD012773062CF3FA2549E44F58ED2401710
+    seckey = bytes_from_int(0x0B432B2677937381AEF05BB02A66ECD012773062CF3FA2549E44F58ED2401710)
     msg = bytes_from_int(0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
     sig = schnorr_sign(msg, seckey)
-    return (bytes_from_int(seckey), pubkey_gen(seckey), msg, sig, "TRUE", "test fails if msg is reduced modulo p or n")
+    return (seckey, pubkey_gen(seckey), msg, sig, "TRUE", "test fails if msg is reduced modulo p or n")
 
-# Signs with a given nonce. Results in an invalid signature if y(kG) is not a
-# quadratic residue.
+# Signs with a given nonce. Results in an invalid signature if y(kG) is not square.
 def schnorr_sign_fixed_nonce(msg, seckey0, k):
```

Comment:
That's a scary function now that I see it. Can we add a big fat warning here, and/or rename this to `insecure_signature_only_for_test_vectors()` or something? Or maybe even better, define a `schnorr_sign` function that calls an external function to generate k properly, and then temporarily overwrite this using __code__ tricks in the test vectors.

Reply:
My very first iteration of the test vector generation code was using `__code__` tricks but I changed it to a more straightforward copy and paste because it was way easier to understand and less trickery. Additionally, it's a bit messy because one differences between the two functions is that in `schnorr_sign_fixed_nonce` the nonce `k` is never negated. I renamed the function name for now and added a warning.